plugins/modules/keycloak_user.py

```diff
@@ -0,0 +1,496 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2019, INSPQ (@elfelip)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: keycloak_user
+short_description: create and Configure a user in Keycloak
+description:
+    - This module creates, removes or update Keycloak users.
+version_added: 7.0.0
+options:
+    realm:
+        description:
+            - The name of the realm in which is the client.
+        default: master
+        type: str
+    username:
+        description:
+            - username for the user.
+        required: true
+        type: str
+    id:
+        description:
+            - ID of the user on the keycloak server if known
+        type: str
+    enabled:
+        description:
+            - Enabled user.
+        default: true
+        type: bool
+    email_verified:
+        description:
+            - check the validity of user email.
+        default: false
+        type: bool
+        aliases:
+            - emailVerified
+    first_name:
+        description:
+            - User firstName.
+        required: false
+        type: str
+        aliases:
+            - firstName
+    last_name:
+        description:
+            - User lastName.
+        required: false
+        type: str
+        aliases:
+            - lastName
+    email:
+        description:
+            - User email.
+        required: false
+        type: str
+    federation_link:
+        description:
+            - Federation Link.
+        required: false
+        type: str
+        aliases:
+            - federationLink
+    service_account_client_id:
+        description:
+            - Description of the client Application.
+        required: false
+        type: str
+        aliases:
+            - serviceAccountClientId
+    client_consents:
+        description:
+            - client Authenticator Type.
+        type: list
+        elements: dict
+        default: []
+        aliases:
+            - clientConsents
+        suboptions:
+            client_id:
+                description:
+                    - Client ID of the client role. Not the technical id of the client.
```

Comment:
Abbreviations should always be uppercase.
```suggestion
                    - Client ID of the client role. Not the technical ID of the client.
```